From 3b550694ad35b5fd7238fa58d5e057767230d72b Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Fri, 5 Sep 2025 23:57:30 +0800
Subject: [PATCH 85/94] ia64: unwind: correct typed argument for UNW_DPRINT()

Fix the following error during build (with CONFIG_WERROR):

arch/ia64/kernel/unwind.c:1645:23: error: format '%lx' expects argument of type 'long unsigned int', but argument 3 has type 'long long unsigned int' [-Werror=format=]
 1645 |         UNW_DPRINT(1, "unwind.%s: state record for func 0x%lx, t=%u:\n",
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1646 |                 __func__, table->segment_base + e->start_offset, sr.when_target);
      |                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                               |
      |                                               long long unsigned int
./include/linux/printk.h:486:25: note: in definition of macro 'printk_index_wrap'
  486 |                 _p_func(_fmt, ##__VA_ARGS__);                           \
      |                         ^~~~
arch/ia64/kernel/unwind.c:66:52: note: in expansion of macro 'printk'
   66 | #  define UNW_DPRINT(n, ...)  if (UNW_DEBUG_ON(n)) printk(__VA_ARGS__)
      |                                                    ^~~~~~
arch/ia64/kernel/unwind.c:1645:9: note: in expansion of macro 'UNW_DPRINT'
 1645 |         UNW_DPRINT(1, "unwind.%s: state record for func 0x%lx, t=%u:\n",
      |         ^~~~~~~~~~
arch/ia64/kernel/unwind.c:1645:61: note: format string is defined here
 1645 |         UNW_DPRINT(1, "unwind.%s: state record for func 0x%lx, t=%u:\n",
      |                                                           ~~^
      |                                                             |
      |                                                             long unsigned int
      |                                                           %llx

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 arch/ia64/kernel/unwind.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/ia64/kernel/unwind.c b/arch/ia64/kernel/unwind.c
index 44620c781ac9..f8b835919516 100644
--- a/arch/ia64/kernel/unwind.c
+++ b/arch/ia64/kernel/unwind.c
@@ -1638,7 +1638,7 @@ build_script (struct unw_frame_info *info)
 	}
 
 #ifdef UNW_DEBUG
-	UNW_DPRINT(1, "unwind.%s: state record for func 0x%lx, t=%u:\n",
+	UNW_DPRINT(1, "unwind.%s: state record for func 0x%llx, t=%u:\n",
 		__func__, table->segment_base + e->start_offset, sr.when_target);
 	for (r = sr.curr.reg; r < sr.curr.reg + UNW_NUM_REGS; ++r) {
 		if (r->where != UNW_WHERE_NONE || r->when != UNW_WHEN_NEVER) {
-- 
2.25.1

